artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConnectionView.java

```diff
@@ -83,12 +83,12 @@ public Object getField(RemotingConnection connection, String fieldName) {
          case "remoteAddress":
             return connection.getRemoteAddress();
          case "users":
-            Set<String> users = new HashSet<>();
+            Set<String> users = new TreeSet<>();
             for (ServerSession session : sessions) {
                String username = session.getUsername() == null ? "" : session.getUsername();
                users.add(username);
             }
-            return users;
+            return StringUtil.joinStringList(users, ",");
```

Comment:
Whilst i understand treeset to give ordering by insert, the return shouldn't change the type, e.g just return users still. Or im missing something.... i guess a test may help to describe/show why its needed.

Reply:
The returned value is a set of users of a single connection. to sort the connections by their user(s), these lists need to be comparable with each other. however treeset does not implement the comparable interface, so you cannot sort multiple treesets by there values. thus the PR now returns a comparable/sortable string representation of the set. does that make sense?

Reply:
The returned value is a set of users of a single connection. to sort the connections by their user(s), these lists need to be comparable with each other. however treeset does not implement the comparable interface, so you cannot sort multiple treesets by there values. thus the PR now returns a comparable/sortable string representation of the set. does that make sense?